frontend/src/contexts/authStateContext.tsx

```diff
@@ -17,16 +17,16 @@ export const AuthStateContextProvider = (props: { children?: any }) => {
 
   React.useEffect(() => {
     const loadUserInfo = async () => {
-      const user = await keycloak.obj?.loadUserInfo();
-      setUserInfo(user);
+      try {
+        const user = await keycloak.obj?.loadUserInfo();
+        setUserInfo(user);
+      } catch (err) {
+        // this error isn't recoverable, so just log it for debugging purposes.
+        console.error(err);
+      }
     };
 
-    try {
-      if (keycloak.obj.authenticated) loadUserInfo();
-    } catch (err) {
-      // this error isn't recoverable, so just log it for debugging purposes.
-      console.error(err);
-    }
+    if (keycloak.obj.authenticated) loadUserInfo();
```

Comment:
A few lines above we protect against a null `keycloak.obj` with `const user = await keycloak.obj?.loadUserInfo();`

Wouldn't it make sense to do it here as well?

```typescript
 if (keycloak.obj?.authenticated) loadUserInfo();
```

Reply:
this loadUserInfo is just an async function defined on line 19. That function guards against a null keycloak.obj in the method you describe.